src/librustc_middle/ty/mod.rs

```diff
@@ -120,12 +120,12 @@ mod sty;
 pub struct ResolverOutputs {
     pub definitions: rustc_hir::definitions::Definitions,
     pub cstore: Box<CrateStoreDyn>,
-    pub extern_crate_map: NodeMap<CrateNum>,
-    pub trait_map: TraitMap<NodeId>,
-    pub maybe_unused_trait_imports: NodeSet,
-    pub maybe_unused_extern_crates: Vec<(NodeId, Span)>,
-    pub export_map: ExportMap<NodeId>,
-    pub glob_map: GlobMap,
+    pub extern_crate_map: FxHashMap<DefId, CrateNum>,
+    pub trait_map: FxHashMap<hir::HirId, Vec<hir::TraitCandidate<hir::HirId>>>,
+    pub maybe_unused_trait_imports: FxHashSet<LocalDefId>,
+    pub maybe_unused_extern_crates: Vec<(DefId, Span)>,
+    pub export_map: ExportMap<hir::HirId>,
+    pub glob_map: FxHashMap<LocalDefId, FxHashSet<Symbol>>,
```

Comment:
It does seem like all of these could plausibly be `DefId`. Did you find something during your experiments that wasn't a definition? If not, we should open a cleanup issue once this gets merged.

Reply:
I've only tried for `trait_map` but yes, for this field, there is no def id, only hir id

Reply:
@petrochenkov I see you've assigned yourself. Any recommendations here? Is there a reason that `trait_map` and `export_map` cannot use `DefId`s?

Reply:
Also because the global context uses `HirId`, using `DefId` here would mean another map clone/conversion in `librustc_middle`.

Reply:
The zoo of `DefId`s, `LocalDefId`s and `HirId`s here is a bit surprising (all of these could be `LocalDefId`s (or `HirId`s)), but it looks like the choices are made to minimize conversions at use sites.

I guess the right thing to do here would be changing `DefId`s into `LocalDefId`s in `GlobalCtxt` which is the end destination of these tables.

Reply:
I can definitely take a look at that in a follow up PR!